src/tscore/DbgCtl.cc

```diff
@@ -129,6 +129,7 @@ DbgCtl::_new_reference(char const *tag)
   TSDbgCtl ctl;
 
   ctl.tag = tag;
+  ctl.on  = diags() && diags()->tag_activated(tag, DiagsTagType_Debug);
```

Comment:
If you set "on" without the Registry mutex locked, there's a race condition, at least hypothetically.  It may not get updated for a change to the regex tags are matched to, to see if they are on.

Reply:
I can revert this patch. The idea was to perform this outside of the lock to minimize where we can deadlock. But it shouldn't been needed to avoid the deadlock.

Reply:
Actually, things still do deadlock if I revert this part of the patch. A later function in the periodic logger tries to Debug log, and sets `ctl.on` via this call while the DbgCtl lock is held and deadlocks.